src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -265,6 +264,20 @@ bool value_set_dereferencet::dereference_type_compare(
   return false;
 }
 
+/// \param what: value set entry to convert to an expression: either
+///   ID_unknown, ID_invalid, or an object_descriptor_exprt giving a referred
+///   object and offset.
+/// \param pointer_expr: pointer expression that may point to `what`
+/// \return a `valuet` object containing `guard`, `value` and `ignore` fields.
+///   The `ignore` field is true for a `null` object when `exclude_null_derefs`
+///   is true and integer addresses in java mode.
+///   The guard is an appropriate check to determine whether `pointer_expr`
+///   really points to `what`.
+///   The value corresponds to the dereferenced pointer_expr assuming it is
+///   pointing to the object described by `what`.
+///   For example, we might return
+///   `{.value = global, .pointer_guard = (pointer_expr == &global),
+///     .ignore = false}`
```

Comment:
Why is this being moved to the implementation?

Reply:
It's a private method so should be considered internal to the module I think.